tests/test_models/test_inference/test_forecast.py

```diff
@@ -800,3 +836,109 @@ def test_forecast_subset_segments(self, model, transforms, example_tsds):
+        test_ts.transform(train_ts.transforms)
+        model.fit(train_ts)
+
+        # forecasting
+        import torch  # TODO: remove after fix at issue-802
+
+        torch.manual_seed(11)
+
+        forecast_ts = test_ts.make_future(future_steps=prediction_size, tail_steps=model.context_size)
+        forecast_ts = make_forecast(model=model, ts=forecast_ts, prediction_size=prediction_size)
+
+        # checking
+        forecast_df = forecast_ts.to_pandas(flatten=True)
+        assert not np.any(forecast_df["target"].isna())
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (CatBoostModelMultiSegment(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (LinearMultiSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (ElasticMultiSegmentModel(), [LagTransform(in_column="target", lags=[5, 6])]),
+            (RNNModel(input_size=1, encoder_length=7, decoder_length=7, trainer_params=dict(max_epochs=1)), []),
+            (
+                MLPModel(input_size=2, hidden_size=[10], decoder_length=7, trainer_params=dict(max_epochs=1)),
+                [LagTransform(in_column="target", lags=[5, 6])],
+            ),
+        ],
+    )
+    def test_forecast_new_segments(self, model, transforms, example_tsds):
+        self._test_forecast_new_segments(example_tsds, model, transforms, train_segments=["segment_1"])
+
+    @pytest.mark.parametrize(
+        "model, transforms",
+        [
+            (
+                DeepARModel(max_epochs=1, learning_rate=[0.01]),
+                [
+                    PytorchForecastingTransform(
+                        max_encoder_length=5,
+                        max_prediction_length=5,
+                        time_varying_known_reals=["time_idx"],
+                        time_varying_unknown_reals=["target"],
+                        target_normalizer=GroupNormalizer(groups=["segment"]),
+                    )
+                ],
+            ),
+            (
+                TFTModel(max_epochs=1, learning_rate=[0.01]),
+                [
+                    PytorchForecastingTransform(
+                        max_encoder_length=21,
+                        min_encoder_length=21,
+                        max_prediction_length=5,
+                        time_varying_known_reals=["time_idx"],
+                        time_varying_unknown_reals=["target"],
+                        static_categoricals=["segment"],
+                        target_normalizer=None,
+                    )
+                ],
+            ),
+        ],
+    )
+    def test_forecast_new_segments_failed_encoding_error(self, model, transforms, example_tsds):
+        with pytest.raises(KeyError, match="Unknown category"):
+            self._test_forecast_new_segments(example_tsds, model, transforms, train_segments=["segment_1"])
+
+    @to_be_fixed(raises=NotImplementedError, match="Per-segment models can't make predictions on new segments")
```

Comment:
How we are going to fix it?

Reply:
I don't really know how. We can potentially make it possible for naive models, but for other models I don't think we can do smth like that.

Reply:
Let's try to make it possible for naive models, if don't come to simple solution may be it is better to skip it